packages/ui/src/components/va-sidebar/VaSidebar.vue

```diff
@@ -93,11 +97,22 @@ export default defineComponent({
       isHovered.value = props.hoverable && newHoverState
     }
 
+    const rootElement = shallowRef<HTMLElement>()
+
+    useClickOutside([rootElement], () => {
+      if (props.closeOnClickOutside && props.modelValue) {
+        setTimeout(() => {
```

Comment:
Why is setTimeout here?

Reply:
It fixes an issue when you try to hide sidebar by clicking to checkbox, in this case we have two sources that change modeValue and Sidebar can't be closed. setTimeout with 0 delay creates macrotask, and this means that close-outside handler does't fire update:modelValue immediately. So, the delay is enough to change modelValue from checkbox source (or button) and after that click-outside handler can't brake right behaviour

Reply:
Maybe add `if modelValue != false  { emit(modelValue, false)` to prevent firing update event twice?